### PR TITLE
New markup improvements

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

File: src/devtools/client/inspector/markup/components/TextNode.js [hunk 1]
```diff
@@ -7,14 +7,14 @@ const { getStr, getFormatStr } = require("../utils/l10n");
...
-      type: PropTypes.string.isRequired,
+      type: PropTypes.number.isRequired,
       value: PropTypes.string.isRequired,
     };
   }
 
   render() {
     const { type, value } = this.props;
-    const isComment = type === "comment";
+    const isComment = type === 8;
```
Comment (jasonLaster) [inline src/devtools/client/inspector/markup/components/TextNode.js:17]: lets use NODE.COMMENT

https://github.com/jasonLaster/devtools/blob/master/src/devtools/shared/dom-node-constants.js#L14

File: src/devtools/client/inspector/markup/new-markup.js [hunk 2]
```diff
@@ -326,7 +326,7 @@ class MarkupView {
       this.expandNode(nodeId);
     }
 
-    this.store.dispatch(updateSelectedNode(nodeId));
+    this.selection.setNodeFront(this.nodes.get(nodeId));
```
Comment (jasonLaster) [inline src/devtools/client/inspector/markup/new-markup.js:329]: why do we not need to update the redux store?

File: src/devtools/client/inspector/markup/components/TextNode.js (hunk 1 above)
Comment (hbenl) [reply src/devtools/client/inspector/markup/components/TextNode.js]: Yes!

File: src/devtools/client/inspector/markup/new-markup.js (hunk 2 above)
Comment (hbenl) [reply src/devtools/client/inspector/markup/new-markup.js:329]: The selection will send a `new-node-front` event, which results in a call to `this.update()`, which will update the redux store.

**Review by jasonLaster — APPROVED** [review]

File: src/devtools/client/inspector/markup/new-markup.js (hunk 2 above)
Comment (jasonLaster) [reply src/devtools/client/inspector/markup/new-markup.js:329]: Nice

**Pull request merged**